src/data/markdown/docs/01 guides/05 Testing Guides/04 Large-scale k6 testing.md

```diff
@@ -0,0 +1,484 @@
+	discardResponseBodies: true,
+};
+```
+</div>
+
+k6 by default loads the response body of the request into memory. This causes much higher memory consumption, and often is completely unnecessary.
+
+### --no-thresholds --no-summary 
+
+If you are running a cloud test with local execution (`k6 run -o cloud`), you may want to disable the terminal summary 
+and local threshold calculation because thresholds and summary will be displayed in the cloud. 
+This will save you some memory and CPU cycles.
+
+Here are all the mentioned flags, all in one:
+
+<div class="code-group">
+
+```bash
+k6 run scripts/website.es5.js \
+  -o cloud \
+  --vus=20000 \
+  --duration=10m \
+  --compatibility-mode=base \
+  --no-thresholds \
+  --no-summary \
+```
+
+</div>
+
+### Remove unnecessary checks, groups and custom metrics
+
+If everything else has failed and you are trying to squeeze more performance out of the hardware, 
+you can consider optimizing the code of the load test itself.
+
+**Checks and groups**
+
+k6 records the result of every individual check and group separately. If you are using many checks and groups, you may consider removing them to boost performance.
+
+**Custom metrics**
+
+Similar to checks, values for custom metrics (Trend, Counter, Gauge and Rate) are recorded separately. Consider minimizing the usage of custom metrics.
+
+**Thresholds with abortOnFail**
+
+If you have configured [abortOnFail thresholds](https://k6.io/docs/using-k6/thresholds#aborting-a-test-when-a-threshold-is-crossed), k6 needs to evaluate the result constantly to verify that the threshold wasn't crossed. Consider removing this setting.
+
+
+
+## File upload testing
+Special considerations must be taken when testing file-uploads. 
+
+### Network throughput
+The network throughput of the load-generator machine, as well as the SUT will likely be the bottleneck. 
+
+### Memory
+k6 needs significant amount of memory when uploading files, as every VU is independent and has its own memory.
+
+### Data transfer costs
+k6 can upload a large amount data in a very short period of time. Make sure you understand the data transfer costs before commencing a large scale test.
+
+[Outbound Data Transfer is expensive in AWS EC2](https://www.cloudmanagementinsider.com/data-transfer-costs-everything-you-need-to-know/). The price ranges between $0.08 to $0.20 per GB depending on the region. 
+If you use the cheapest region the cost is about $0.08 per GB. Uploading 1TB therefore costs about $80. Long running test can cost several hundreds of dollars in data transfer alone.
+
+### EC2 costs
+The AWS EC2 instances are relatively cheap. Even the largest instance we have used in this benchmark (m5.24xlarge) costs only $4.6 per hour. Make sure to turn off the load-gen servers once you are done with your testing. Forgotten EC2 server will cost $3312 per month.  
+Tip: it's often possible to launch "spot instances" of the same hardware for 10-20% of the cost. 
+
+## Errors
```

Comment:
This section is really good, but it feels like it shouldn't be in its own article on "Understanding k6 errors" or something like that.

Reply:
@robingustafsson I suspect there's a negation too much in that sentence? 😅  If so, I agree that it would be nice to have in its own article.

Reply:
:+1: I'll release this as is, and then branch it out to another article :-)